### PR TITLE
More Machine Blessing Options

### PR DESCRIPTION
**Pull request by Zenog400**
## About The Pull Request

Adds more implants to the pool that the Machine Blessing rite can draw from, and rebalances the drop rates so that you're more likely to get thematically-consistent implants.

## Why It's Good For The Game

Currently, the options for what the Machine Blessing rite can give you are kinda garbage, considering that _one_ implant from the rite is the same value as _two_ full species conversions. This adds a few more that are actually useful.

Specific implants added:
- Combat arm (extremely low drop rate)
- Tool arm (higher drop rate)
- Reviver implant (lower drop rate)
- Anti-drop implant
- Anti-stun implant (significantly lower drop rate)
- Welding eyes

Additionally, slightly lowers the drop rate of the breathing tube implant, because androids don't have lungs

## Changelog

:cl:
add: increased the size of the pool that Machine Blessing draws from
balance: weighted some of the implants that Machine Blessing can give
/:cl:

### CODE REVIEW

**Comment by Zenog400** [comment]
It's been about 5 years since I coded in DM and I don't really remember how to do it, so if anyone could figure out why it's throwing errors, that'd be super cool

**Comment by Jacquerel** [comment]
use `pick_weight` instead of whatever it is you are doing

**Comment by Time-Green** [comment]
Please add the implants to the PR description so people can see what you’re adding without looking at code

**Comment by Zenog400** [comment]
Okay so apparently the documentation was lying (or I was using the wrong line break), and you need to use `;` to add relative probabilities to `pick`, and using line breaks doesn't work.

**Comment by jughu** [comment]
Could you just split the combat arm implant up in their seperate parts instead of it being a medibeam/flash/esword/stunarm in one package so instead you have 4 extra implants instead of it all being a single implant which would increase the loot pool and make the implant not feel as gamey

**Comment by Zenog400** [comment]
> Could you just split the combat arm implant up in their seperate parts instead of it being a medibeam/flash/esword/stunarm in one package so instead you have 4 extra implants instead of it all being a single implant which would increase the loot pool and make the implant not feel as gamey

I mean, combat arm has _roughly_ the same drop rate as a pulse rifle from an arcade machine (slightly lower if I understand the code for `pick` right), and it's 2000 favor for every roll instead of just two arcade tickets.

**Comment by Loafin34** [comment]
flashlight eyes and cybernetic moth eyes please

File: code/modules/religion/rites.dm [hunk 1]
```diff
@@ -129,11 +129,17 @@
...
 		/obj/item/organ/internal/cyberimp/eyes/hud/medical,
-		/obj/item/organ/internal/cyberimp/mouth/breathing_tube,
+		prob(80);/obj/item/organ/internal/cyberimp/mouth/breathing_tube,
+		prob(50);/obj/item/organ/internal/cyberimp/chest/reviver,
 		/obj/item/organ/internal/cyberimp/chest/thrusters,
+		/obj/item/organ/internal/cyberimp/brain/anti_drop,
+		prob(10);/obj/item/organ/internal/cyberimp/brain/anti_stun,
+		/obj/item/organ/internal/eyes/robotic/shield,
 		/obj/item/organ/internal/eyes/robotic/glow,
 	)
```
Comment (LemonInTheDark) [inline code/modules/religion/rites.dm:144]: use pick_weight, please for the love of god and man use pick_weight

**Comment by LemonInTheDark** [comment]
Also is this something you're capable of testing? I'll let you off cause it's just line by lines but you should really have a local setup

File: code/modules/religion/rites.dm (hunk 1 above)
Comment (ZephyrTFA) [reply code/modules/religion/rites.dm:144]: holy shit
absolutely not

Comment (Zenog400) [reply code/modules/religion/rites.dm:144]: pick_weight isn't in the documentation, I used what I could find

Comment (LemonInTheDark) [reply code/modules/religion/rites.dm:144]: it's in code/__HELPERS/_lists.dm
we wrote it ourselves

image

Can always ask in discord if you're unsure of stuff like this (or check other places that do similar work) if you're unsure, many of the helpers we use aren't inbuilt.

Comment (LemonInTheDark) [reply code/modules/religion/rites.dm:144]: (format is a list of things to pick paired with the weights to pick them out of the sum total)
image

**Comment by Zenog400** [comment]
Fuck me how do you define a variable in dm?

**Comment by Zenog400** [comment]
Fuck me, how do you _call_ a variable in dm?

**Comment by Jacquerel** [comment]
> Fuck me, how do you _call_ a variable in dm?

You can only use static data in the default values in a class declaration, if you want to construct a list composed of other list vars on the item you'd have to do it within a proc (probably initialize)

File: code/modules/religion/rites.dm [hunk 2]
```diff
@@ -128,13 +128,31 @@
...
+	var/head = list(
+		/obj/item/organ/internal/cyberimp/brain/anti_drop = 100,
+		/obj/item/organ/internal/cyberimp/brain/anti_stun = 10
+	)
+	var/blessing = pick_weight_recursive(
+	arm = 15,
+	eyes = 15,
+	chest = 9,
+	head = 10,
+	/obj/item/organ/internal/cyberimp/mouth/breathing_tube = 5
```
Comment (Jacquerel) [inline code/modules/religion/rites.dm:155]:
```suggestion
	list(
		arm = 15,
		eyes = 15,
		chest = 9,
		head = 10,
		/obj/item/organ/internal/cyberimp/mouth/breathing_tube = 5,
	)
```

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/modules/religion/rites.dm (hunk 2 above)
Comment (Jacquerel) [reply code/modules/religion/rites.dm:155]: Pick_weight reads from a list as its first argument.
You're instead providing 5 arguments and it assumes the first four are an attempt at calling named arguments.

**Comment by Zenog400** [comment]
> Also is this something you're capable of testing? I'll let you off cause it's just chance changes but you should really have a local setup

I don't have my local setup anymore, unfortunately, the computer it was on died a couple years ago. If I actually start trying to code things that really _do_ something instead of just changing a loot table, though, I'll definitely set a new one up.

**Pull request merged**